File: traslate_py_C/example4.c
#include <stdio.h>
#include <math.h>

int main(void){
    int a, b, x;
    
    printf("Valor de a: ");
    scanf("%d", &a);
    printf("Valor de b: ");
    scanf("%d", &b);
    
    if (a != 0){
        x = -b/a;
        printf("solucion %d\n", x);
    } else if (b != 0){
        printf("La ecuacion no tiene solucion\n");
    } else {
        printf("La ecuacion tiene infinitas soluciones.\n");
    }
    
    return 0;
}

File: traslate_py_C/example4.py
a = float(raw_input("Valor de a: "))
b = float(raw_input("Valor de b: "))

if a != 0:
    x = -b/a
    print "Solucion: ", x
else:
    if b != 0:
        print "La ecuacion no tiene solucion."
    else:
        print "La ecuacion tiene infinitas soluciones."